src/eos-updater-poll.c

```
@@ -17,199 +18,279 @@
+          !g_error_matches (local_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_NOT_FOUND))
         {
-          eos_updater_set_full_download_size (updater, -1);
-          eos_updater_set_full_unpacked_size (updater, -1);
-          eos_updater_set_download_size (updater, -1);
-          eos_updater_set_unpacked_size (updater, -1);
-          eos_updater_set_downloaded_bytes (updater, -1);
-
-          /* shouldn't actually stop us offering an update, as long
-           * as the branch itself is resolvable in the next step,
-           * but log it anyway.
-           */
-          if (error)
-            {
-              message ("No size summary data: %s", error->message);
-              g_clear_error (&error);
-            }
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return FALSE;
         }
+
+      /* Config file was not found, fall back to the defaults
+       */
+      sources_config->download_order = g_array_sized_new (FALSE, /* not null terminated */
+                                                          FALSE, /* no clearing */
+                                                          sizeof (EosUpdaterDownloadSource),
+                                                          1);
+      g_array_append_val (sources_config->download_order, main_source);
+      sources_config->volume_path = NULL;
+
+      return TRUE;
     }
-  else /* csum == NULL means OnHold=true, nothing to do here */
-    eos_updater_set_state_changed (updater, EOS_UPDATER_STATE_READY);
 
- out:
-  if (error)
+  download_order_strv = g_key_file_get_string_list (config,
+                                                    DOWNLOAD_GROUP,
+                                                    ORDER_KEY,
+                                                    NULL,
+                                                    error);
+  if (download_order_strv == NULL)
+    return FALSE;
```

I think it should error out here, since the user has provided an invalid configuration. With c29741b, there is now an example configuration file installed in `/usr/share/eos-updater/etc/eos-updater.conf` which the user can copy, and which is loaded if no configuration file exists in `/etc`, but not if loading an existing file fails.